### ISSUE TITLE
typing_extensions.TypedDict no longer recognised

### ISSUE THREAD

**Issue by Terrance** [microsoft/pyright#2193]
## Describe the bug

`TypedDict`, when imported from `typing_extensions`, is being flagged with _Argument to class must be a base class_ on use.

This used to work fine, so is likely to be a regression -- possibly related to #2174?  It doesn't appear to be due to a change with `typing_extensions`, older versions are also affected.  When imported via `typing` there are no issues.

## To reproduce

```python
# (no problems here)

from typing import TypedDict

class Data(TypedDict):
    field: int
```

```python
try:
    from typing import TypedDict
except ImportError:
    from typing_extensions import TypedDict

class Data(TypedDict):  # Argument to class must be a base class
    field: int
```

## VS Code extension or command-line

Pyright 1.1.162 in VSCode.

**Comment by erictraut** [comment]
It works fine if you always import from `typing_extensions`. The problem is that you are importing two versions within the `try` block, so the effective type of the symbol is a union between two different classes. 

Your options are:
1. Remove the `try` block and always import `TypedDict` from `typing_extensions`, at least until you remove support for Python versions older than Python 3.8
2. Use an `if not TYPE_CHECKING` statement to mask one of the two definitions.

```python
try:
    from typing import TypedDict
except ImportError:
    if not TYPE_CHECKING:
        from typing_extensions import TypedDict
```

**Comment by Terrance** [comment]
So it does -- thought I'd tried it outside of the `try` block but apparently not...

The `TYPE_CHECKING` pattern is a bit messy, particularly if multiple backport libraries are involved (though at least I'm not using any others that require subclassing) across multiple files.  In this case I can just revert to always installing `typing_extensions`, seeing as it's a light enough dependency and it'll proxy the builtins on newer versions.

Sadly I'll be supporting Python 3.6 for a while yet (with the likes of Ubuntu 18.04 LTS still providing it).
